contexts/WalletManagerContext.tsx

```diff
@@ -593,7 +591,7 @@ export function WalletManagerProvider({ children }: { children: ReactNode }) {
 
   // Refresh accounts after settings changed
   useEffect(() => {
-    if (!rpcClient) {
+    if (!rpcClient || isWalletSettingsLoading) {
```

Comment:
The `useEffect` dependency should add `isWalletSettingLoading` as well.

Reply:
c28d67b0f1240fbc0bdc8a403eedada847e5ee73